web_field_required_invisible_manager/models/models.py

```diff
@@ -27,34 +27,32 @@ class IrModel(models.Model):
     )
 
 
-class Base(models.AbstractModel):
-    _inherit = "base"
+class IrUiView(models.Model):
+    _inherit = "ir.ui.view"
 
-    @api.model
-    def fields_view_get(
-        self, view_id=None, view_type=False, toolbar=False, submenu=False
-    ):
-        arch = super().fields_view_get(
-            view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu
-        )
-        if view_type not in ["form", "tree"]:
-            return arch
-        # TODO speed up somehow
+    def postprocess_and_fields(self, node, model=None, validate=False):
+        arch, new_fields = super().postprocess_and_fields(node, model, validate)
+        if self.type not in ["form", "tree"] and node.tag not in ["form", "tree"]:
+            return arch, new_fields
         restrictions = self.env["custom.field.restriction"].search(
             [
-                ("model_name", "=", self._name),
+                ("model_name", "=", self.model or model),
```

Comment:
**issue**: from base Odoo, it looks like `model` should have priority over `self.model`

https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_ui_view.py#L857